src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs

```diff
@@ -183,7 +183,7 @@ private static bool CheckScoreColumnKind(Schema schema, int col)
 
             // Get the score column set id from colScore.
             var type = schema[colScore].Metadata.Schema.GetColumnOrNull(MetadataUtils.Kinds.ScoreColumnSetId)?.Type;
-            if (type == null || !(type is KeyType) || type.RawKind != DataKind.U4)
+            if (type == null || !(type is KeyType) || type.RawType != typeof(uint))
```

Comment:
>type == null || [](start = 16, length = 15)

Similar note, I think we can do without the null check given that the `is` test will catch the null case anyway, I think?

Reply:
Fixed.

---
In reply to: [248939217](https://github.com/dotnet/machinelearning/pull/2176#discussion_r248939217) [](ancestors = 248939217)